source/libraries.md

```diff
@@ -1,22 +1,34 @@
 (c:impl.libs)=
-# Libraries
+# Code Packages
 
-There are reference implementations of PALS parsers.
+There are reference implementations of PALS "parsers". Here the word "parser" is used loosely since
+the parsers do much more than parse lattice files. All parsers will:
 
+* Write and read from/to files,
+* Expand & evaluate expressions on lattices ( [`lattice expansion`](#s:lattice.expand)),
+* Validate existing files.
```

Comment:
I do not think that all parsers _must validate_ a PALS file. Some, say a Fortran implementation, might simply assume a PALS input is valid and rely on an external validator tool to diagnose import issues. Others, e.g., a scripted implementation, can give user-friendly an detailed hints when they discover invalid parts of input.

Reply:
Open suggestion :)